mlrun/api/initial_data.py

```diff
@@ -259,6 +281,27 @@ def _add_default_marketplace_source_if_needed(
     return
 
 
+def _add_data_version(
+    db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
+):
+    if db.get_version(db_session, data_version_name, raise_on_not_found=False) is None:
+        projects = db.list_projects(db_session)
+        # heuristic - if there are no projects it's a new DB - data version is latest
+        if not projects.projects:
+            logger.info(
+                "Setting data version to latest",
+                latest_data_version=latest_data_version,
+            )
+            db.create_version(db_session, data_version_name, str(latest_data_version))
+        else:
+            # This code was added to 0.8.0
+            # The latest data migration added before adding this was added back in 0.6.0
+            # We can safely assume no one is upgrading from a version pre 0.6.0 directly to 0.8.0, therefore we assume
+            # the data already passed version 0 to 1 migrations
+            logger.info("Setting data version to 1")
```

Comment:
No
This is the scenario where I didn't find the version of the data in the DB, but there is data in the DB (meaning it's an upgrade an not empty DB)
What I can assume is only that this data passed migrations to version 1.
let's assume for 0.8.1 I'll add a new migration - then I'll bump `latest_data_version` to 2, if someone will upgrade from 0.6.5 to 0.8.1 their data passed version 1 migrations, but not version 2, therefore I want to tag it as version 1, and then it will pass migrations to version 2, and then version will be updated.